File: app/src/main/java/com/smedialink/abakarmagomedov/githubapplication/di/AppModule.java
package com.smedialink.abakarmagomedov.githubapplication.di;

import android.content.Context;
import android.support.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by abakarmagomedov on 15/05/17.
 */

@Singleton
@Module
public class AppModule {

    private final Context mContext;

    public AppModule(Context context) {
        mContext = context;
    }

    @Singleton
    @NonNull
    @Provides
    Context provideContext() {
        return mContext;
    }


}

File: app/src/main/java/com/smedialink/abakarmagomedov/githubapplication/di/AppModule.kt
package com.smedialink.abakarmagomedov.githubapplication.di

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by abakarmagomedov on 15/05/17.
 */
@Singleton
@Module
class AppModule(private val mContext: Context) {
    @Singleton
    @Provides
    fun provideContext(): Context {
        return mContext
    }
}
